File urls are missing when logging in a User via authData.

I have two ParseFiles in my Parse.User class. ProfilePic and CoverPhoto.

I've started migration over to Compose.io DBaaS. I've got Parse-Server up and running locally pointing to Compose. That's all working and I can login with a simple test user.

When logging in with a user who does NOT have any old files attached to _User (test user) the login is successful. However, when logging into a _User who still has these old files (on parse s3?) I get this error:

```
[ERROR] :  Script Error Tried to save an object containing an unsaved file.
```

```
//Server side, with verbose=1 returns the full object
....
  "coverPhoto": {
    "__type": "File",
    "name": "tfss-2f7fb0bd-d9e6-4ec0-b20b-80405491afcb-cover.jpg"
  },
...
```


If I delete the coverPhoto key in the database the error goes away so I know it's the old files causing it.

I'm guessing fileKey is supposed to grabbing the file? And it's not? There's no failure in the server side. I have confirmed that the fileKey option is set correctly and and appId matches Parse.com.

Thanks

--- a/spec/ParseUser.spec.js
+++ b/spec/ParseUser.spec.js
@@ -82,8 +82,6 @@ describe('Parse.User testing', () => {
   });
 
   it("user login with files", (done) => {
-    "use strict";
-
     let file = new Parse.File("yolo.txt", [1,2,3], "text/plain");
     file.save().then((file) => {
       return Parse.User.signUp("asdf", "zxcv", { "file" : file });
@@ -927,6 +925,29 @@ describe('Parse.User testing', () => {
         ok(false, "linking should have worked");
         done();
       }
+    });
+  });
+
+  it('log in with provider with files', done => {
+    let provider = getMockFacebookProvider();
+    Parse.User._registerAuthenticationProvider(provider);
+    let file = new Parse.File("yolo.txt", [1, 2, 3], "text/plain");
+    file.save().then(file => {
+      let user = new Parse.User();
+      user.set('file', file);
+      return user._linkWith('facebook', {});
+    }).then(user => {
+      expect(user._isLinked("facebook")).toBeTruthy();
+      return Parse.User._logInWith('facebook', {});
+    }).then(user => {
+      let fileAgain = user.get('file');
+      expect(fileAgain.name()).toMatch(/yolo.txt$/);
+      expect(fileAgain.url()).toMatch(/yolo.txt$/);
+    }).then(() => {
+      done();
+    }, error => {
+      fail(error);
+      done();
     });
   });
 

--- a/src/RestWrite.js
+++ b/src/RestWrite.js
@@ -74,6 +74,8 @@ RestWrite.prototype.execute = function() {
   }).then(() => {
     return this.transformUser();
   }).then(() => {
+    return this.expandFilesForExistingObjects();
+  }).then(() => {
     return this.runDatabaseOperation();
   }).then(() => {
     return this.handleFollowup();
@@ -702,6 +704,16 @@ RestWrite.prototype.handleInstallation = function() {
     // TODO: Validate ops (add/remove on channels, $inc on badge, etc.)
   });
   return promise;
+};
+
+// If we short-circuted the object response - then we need to make sure we expand all the files,
+// since this might not have a query, meaning it won't return the full result back.
+// TODO: (nlutsenko) This should die when we move to per-class based controllers on _Session/_User
+RestWrite.prototype.expandFilesForExistingObjects = function() {
+  // Check whether we have a short-circuited response - only then run expansion.
+  if (this.response && this.response.response) {
+    this.config.filesController.expandFilesInObject(this.config, this.response.response);
+  }
 };
 
 RestWrite.prototype.runDatabaseOperation = function() {

